Terraform crashes when running terraform init on version 1.9.3-1 in arch linux

### Terraform Version

```shell
Terraform v1.9.3
on linux_amd64
```


### Terraform Configuration Files

 ------- seperate FILE -----

```
terraform {
        required_providers {
                docker = {
                        source="kreuzwerker/docker"
                        version = "3.0.2"
                }
        }
}
```
 ------- seperate FILE -----
```
resource "docker_container" "chat_backend" {
        name = "chat_backend"
        image = docker_image.chat_backend.image_id
        env = {
                ENV="development"
        }
        volumes = {
                container_path = "/app"
                host_path = "../"
                volume_name = "chat-backend"
        }
        networks_advanced = {
                name = "chat_network"
        }
        ports = {
                internal = 3000
                external = 3000
        }
}

resource "docker_container" "chat_database" {
        name = "chat_database"
        image = docker_image.chat_database.image_id
        volumes = {
                container_path = "/data"
                host_path = "../chat_database"
                volume_name = "chat-database"
        }
        command = "redis-server"
        networks_advanced = {
                name = "chat_network"
        }
        ports = {
                internal = 3001
                external = 3001
        }
}
```
 ------- seperate FILE -----
```
terraform {
        required_providers {
                docker = {
                        source="kreuzwerker/docker"
                        version="3.0.2"
                }
        }
}

resource "docker_image" "chat_backend" {
        name = "chat-backend"
        build {
                context = "../"
        }
}

resource "docker_image" "chat_database" {
        name = "redis"
        image = "redis:7.4"
}
```
 --------------- Seperate File -----------
```
terraform {
        required_providers {
                docker = {
                        source = "kreuzwerker/docker"
                        version = "3.0.2"
                {
        }
}

resource "docker_network" "chat_network" {
        name = "chat_network"
        driver = "bridge"
}
resource "docker_volume" "chat_database" {
        name = "chat_database"
}

resource "docker_volume" "chat_backend" {
        name = "chat_backend"
}
```


### Debug Output

```
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
Please report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

panic: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:24 +0x5e
github.com/hashicorp/terraform/internal/logging.PanicHandler()
        github.com/hashicorp/terraform/internal/logging/panic.go:84 +0x18b
panic({0x575ecc3a7800?, 0x575ece1d3a40?})
        runtime/panic.go:770 +0x132
github.com/hashicorp/terraform/internal/configs.NewModule({0xc000593ec0, 0x4, 0x1?}, {0x0, 0x0, 0x1?})
        github.com/hashicorp/terraform/internal/configs/module.go:142 +0xa90
github.com/hashicorp/terraform/internal/configs.NewModuleWithTests(...)
        github.com/hashicorp/terraform/internal/configs/module.go:102
github.com/hashicorp/terraform/internal/configs.(*Parser).LoadConfigDirWithTests(0xc0005932e0, {0x575ecc001c80, 0x1}, {0x575ecb768f9f?, 0x0?})
        github.com/hashicorp/terraform/internal/configs/parser_config_dir.go:73 +0x39b
github.com/hashicorp/terraform/internal/command.(*Meta).loadSingleModuleWithTests(0xc0006028c0, {0xc000084054?, 0xc000646f60?}, {0x575ecb768f9f, 0x5})
        github.com/hashicorp/terraform/internal/command/meta_config.go:108 +0xac
github.com/hashicorp/terraform/internal/command.(*InitCommand).Run(0xc0006028c0, {0xc000074050?, 0x0?, 0x0?})
        github.com/hashicorp/terraform/internal/command/init.go:163 +0xe7e
github.com/hashicorp/cli.(*CLI).Run(0xc000203180)
        github.com/hashicorp/cli@v1.1.6/cli.go:265 +0x5b8
main.realMain()
        github.com/hashicorp/terraform/main.go:339 +0x1e6b
main.main()
        github.com/hashicorp/terraform/main.go:64 +0x13

```

### Expected Behavior

TErraform init should have successfully completed without any isssue and all the necessary provider packages should have been installed.

### Actual Behavior

Presented with the error mentione

### Steps to Reproduce

Step 1: Create terraform files given in the terraform config files section
Step 2: Terraform init

I faced this issue in arch linux.

### Additional Context

_No response_

### References

_No response_

--- a/internal/command/init_test.go
+++ b/internal/command/init_test.go
@@ -2949,6 +2949,66 @@ versions are specified, run the following command:
 	}
 }
 
+func TestInit_testsWithOverriddenInvalidRequiredProviders(t *testing.T) {
+	td := t.TempDir()
+	testCopyDir(t, testFixturePath("init-with-overrides-and-duplicates"), td)
+	defer testChdir(t, td)()
+
+	provider := applyFixtureProvider() // We just want the types from this provider.
+
+	providerSource, close := newMockProviderSource(t, map[string][]string{
+		"hashicorp/test": {"1.0.0"},
+	})
+	defer close()
+
+	ui := new(cli.MockUi)
+	view, done := testView(t)
+	c := &InitCommand{
+		Meta: Meta{
+			testingOverrides: metaOverridesForProvider(provider),
+			Ui:               ui,
+			View:             view,
+			ProviderSource:   providerSource,
+		},
+	}
+
+	args := []string{}
+	code := c.Run(args) // just make sure it doesn't crash.
+	if code != 1 {
+		t.Fatalf("expected failure but got: \n%s", done(t).All())
+	}
+}
+
+func TestInit_testsWithInvalidRequiredProviders(t *testing.T) {
+	td := t.TempDir()
+	testCopyDir(t, testFixturePath("init-with-duplicates"), td)
+	defer testChdir(t, td)()
+
+	provider := applyFixtureProvider() // We just want the types from this provider.
+
+	providerSource, close := newMockProviderSource(t, map[string][]string{
+		"hashicorp/test": {"1.0.0"},
+	})
+	defer close()
+
+	ui := new(cli.MockUi)
+	view, done := testView(t)
+	c := &InitCommand{
+		Meta: Meta{
+			testingOverrides: metaOverridesForProvider(provider),
+			Ui:               ui,
+			View:             view,
+			ProviderSource:   providerSource,
+		},
+	}
+
+	args := []string{}
+	code := c.Run(args) // just make sure it doesn't crash.
+	if code != 1 {
+		t.Fatalf("expected failure but got: \n%s", done(t).All())
+	}
+}
+
 func TestInit_testsWithModule(t *testing.T) {
 	// Create a temporary working directory that is empty
 	td := t.TempDir()

--- a/internal/command/testdata/init-with-duplicates/primary.tf
+++ b/internal/command/testdata/init-with-duplicates/primary.tf
@@ -1,0 +1,12 @@
+terraform {
+  required_providers {
+    test = {
+        source = "hashicorp/test"
+        version = "1.0.0"
+    }
+  }
+}
+
+resource "test_instance" "foo" {
+  ami = "bar"
+}

--- a/internal/command/testdata/init-with-duplicates/secondary.tf
+++ b/internal/command/testdata/init-with-duplicates/secondary.tf
@@ -1,0 +1,12 @@
+terraform {
+  required_providers {
+    test = {
+      source = "hashicorp/test"
+      version = "1.0.0"
+    { // This typo is deliberate, we want to test that the parser can handle it.
+  }
+}
+
+resource "test_instance" "bar" {
+  ami = "foo"
+}

--- a/internal/command/testdata/init-with-overrides-and-duplicates/primary.tf
+++ b/internal/command/testdata/init-with-overrides-and-duplicates/primary.tf
@@ -1,0 +1,12 @@
+terraform {
+  required_providers {
+    test = {
+        source = "hashicorp/test"
+        version = "1.0.0"
+    }
+  }
+}
+
+resource "test_instance" "foo" {
+  ami = "bar"
+}

--- a/internal/command/testdata/init-with-overrides-and-duplicates/secondary.tf
+++ b/internal/command/testdata/init-with-overrides-and-duplicates/secondary.tf
@@ -1,0 +1,12 @@
+terraform {
+  required_providers {
+    test = {
+      source = "hashicorp/test"
+      version = "1.0.0"
+    }
+  }
+}
+
+resource "test_instance" "bar" {
+  ami = "foo"
+}

--- a/internal/command/testdata/init-with-overrides-and-duplicates/secondary_override.tf
+++ b/internal/command/testdata/init-with-overrides-and-duplicates/secondary_override.tf
@@ -1,0 +1,12 @@
+terraform {
+  required_providers {
+    test = {
+      source = "hashicorp/test"
+      version = "1.0.0"
+    { // This typo is deliberate, we want to test that the parser can handle it.
+  }
+}
+
+resource "test_instance" "bar" {
+  ami = "override"
+}

--- a/internal/configs/parser_config.go
+++ b/internal/configs/parser_config.go
@@ -120,7 +120,9 @@ func parseConfigFile(body hcl.Body, diags hcl.Diagnostics, override, allowExperi
 				case "required_providers":
 					reqs, reqsDiags := decodeRequiredProvidersBlock(innerBlock)
 					diags = append(diags, reqsDiags...)
-					file.RequiredProviders = append(file.RequiredProviders, reqs)
+					if reqs != nil {
+						file.RequiredProviders = append(file.RequiredProviders, reqs)
+					}
 
 				case "provider_meta":
 					providerCfg, cfgDiags := decodeProviderMetaBlock(innerBlock)

